test_src/sljitTest.c

```diff
@@ -11727,6 +11731,11 @@ static void test92(void)
 	/* buf[33] */
 	sljit_emit_op_flags(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S0), 33 * sizeof(sljit_sw), SLJIT_ATOMIC_STORED);
 
+	sljit_emit_op2(compiler, SLJIT_ADD, SLJIT_R1, 0, SLJIT_S0, 0, SLJIT_IMM, 34 * sizeof(sljit_sw));
+	sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0, SLJIT_MEM1(SLJIT_S0), 32 * sizeof(sljit_sw));
+	sljit_emit_atomic_load(compiler, SLJIT_MOV, SLJIT_R0, SLJIT_R1);
+	sljit_emit_atomic_store(compiler, SLJIT_MOV, SLJIT_R2, SLJIT_R1, SLJIT_R0);
```

Comment:
This can be unsuccessful, and that is valid.

Reply:
I mainly added it for helping with future implementations, and to make sure that the "no flag passed" codepath is covered.  While it could fail, it would be very unlikely as part of sljit_test and might be removed once all other implementations are completed.

Reply:
Flags may be set even if you don't request to set them. For atomics, not setting flag is quite pointless, so I was lazy and set it all the time. But this test is ok as long as the result is not checked.

Reply:
I have an idea here. You could do a regular memory load after the atomic store, and if the value does not match, go back. No thread will overwrite the value, so it should be good enough for testing.

Reply:
Doing instead a byte that is always atomic and can't be perturbed as there is no conflicting access.